packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -872,11 +871,18 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
       }
     }
 
+    // if engine version is pinned and autoMinorVersionUpgrade is not passed in props, we set this property to false
+    let autoMinorVersionUpgrade = props.autoMinorVersionUpgrade;
+    if (props.engine.engineVersion !== undefined && props.autoMinorVersionUpgrade === undefined) {
+      autoMinorVersionUpgrade = false;
+    }
+
     this.instanceType = props.instanceType ?? ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.LARGE);
 
     const instanceParameterGroupConfig = props.parameterGroup?.bindToInstance({});
     this.sourceCfnProps = {
       ...this.newCfnProps,
+      autoMinorVersionUpgrade,
```

Comment:
Why did you move this here?

Reply:
autoMinorVersionUpgrade is present in DatabaseInstanceNewProps and engine is present in DatabaseInstanceSourceProps.
So I deferred setting autoMinorVersionUpgrade and updated it in DatabaseInstanceSource constructor.